cupy/core/_routines_math.pyx

```diff
@@ -424,8 +527,12 @@ cdef ndarray scan(ndarray a, op, dtype=None, ndarray out=None):
 
     cdef int src_cont = int(a._c_contiguous)
     cdef int out_cont = int(out._c_contiguous)
-    kern_scan = _inclusive_scan_kernel(a.dtype, dtype, block_size, op,
-                                       src_cont, out_cont)
+    if numpy.dtype(dtype).char in 'iIlLfd':
```

Comment:
Just a very quick drive-by comment as I noticed something:
```suggestion
    if numpy.dtype(dtype).char in 'iIlLqQfd' and not runtime._is_hip_environment:
```
1. On Windows, `l` is int32 while `q` is int64, and likewise for the uppercases.
2. On HIP, warp shuffle intrinsics are not supported yet 🤬 (https://github.com/ROCm-Developer-Tools/HIP/issues/1491)

Reply:
Thanks as always! Fixed in the way you suggested.